pkgs/pkgs-lib/tests/formats.nix

```
@@ -31,12 +28,61 @@ let
     fi
   '';
 
-  runBuildTests = tests: pkgs.linkFarm "nixpkgs-pkgs-lib-format-tests" (mapAttrsToList (name: value: { inherit name; path = runBuildTest name value; }) (filterAttrs (name: value: value != null) tests));
+  # use this to check for proper serialization
+  shouldPass = format: input: expected: name: {
```

```suggestion
  shouldPass = { format, input, expected }: name: {
```

This does make it a bit more verbose, but adds a lot of clarity about which argument means what. Furthermore it makes adding future arguments easier if necessary.